### DISCUSSION TITLE
return  401 instead of redirect

### DISCUSSION THREAD

**Discussion by vijaypothireddy** [OpenIDC/mod_auth_openidc#653] [Q&A]
I am trying to implement OIDCUnAuthAction  option.

i want to return 401

if it is XMLHttpRequest or 
if the request is not text/html or
if the request is  /assets or /recline or /test or /src

OIDCUnAuthAction 401 "%{REQUEST_URI} =~ m#^/assets\*$#  || %{REQUEST_URI} =~ m#^/recline\*$# || %{REQUEST_URI} =~ m#^/test\*$#  || %{REQUEST_URI} =~ m#^/src\*$# || %{HTTP:X-Requested-With} == 'XMLHttpRequest' || ( ( %{HTTP_ACCEPT} !~ m#text/html# ) && ( %{HTTP_ACCEPT} !~ m#application/xhtml\+xml# ) && ( %{HTTP_ACCEPT} !~ m#\*/\*# ) )"

but when i request resource  under /recline  i get redirect .

my reverse proxy conf

<IfModule mod_ssl.c>
<VirtualHost *:443>
    
....

        ProxyPass "/" "https://<backendip>/"
        ProxyPassReverse "/" "https://<backendip>"
        ProxyPreserveHost On
        SSLProxyVerify none
        SSLProxyCheckPeerCN off
        SSLProxyCheckPeerName off
        SSLProxyCheckPeerExpire off

        OIDCProviderMetadataURL <retracted>
        OIDCClientID   <retracted>
        OIDCClientSecret  <retracted>
        OIDCScope "openid name email"
        OIDCRedirectURI https://<retracted>/redirect_uri/
        OIDCSessionInactivityTimeout 1200
        OIDCStateMaxNumberOfCookies 10 true
        OIDCUnAuthAction 401 "%{REQUEST_URI} =~ m#^/assets\*$#  || %{REQUEST_URI} =~ m#^/recline\*$# || %{REQUEST_URI} =~ m#^/test\*$#  || %{REQUEST_URI} =~ m#^/src\*$# || %{HTTP:X-Requested-With} == 'XMLHttpRequest' || ( ( %{HTTP_ACCEPT} !~ m#text/html# ) && ( %{HTTP_ACCEPT} !~ m#application/xhtml\+xml# ) && ( %{HTTP_ACCEPT} !~ m#\*/\*# ) )"
        OIDCCryptoPassphrase  <retracted>

        <Location />
                AuthType openid-connect
                Require claim <retracted>
        </Location>

</VirtualHost>
</IfModule>

can some one explain whats wrong with syntax or config ?
~

**Comment by zandbelt** [comment]
just use Location directives for /assets or /recline or /test or /src as in:
```
<Location /assets>
  OIDCUnAuthAction 401
</Location>
```

**Reply by vijaypothireddy** [reply]
thanks for quick response.
